### PR TITLE
Update survey abilities

### PR DESCRIPTION
**Pull request by kxmbrian**
This PR updates the abilities for survey to depend on the newly added booleans. The different cases for the abilities are enumerated [here](https://docs.google.com/spreadsheets/d/1rLt9O3K3VI90hrYRgEPeFDudxYGW0_SrLE0gv2N7F1U/edit#gid=0).

This only hides/disables unavailable actions in the frontend. The backend checks will come in the next PR.

### CODE REVIEW

**Comment by weiqingtoh** [comment]
@kxmbrian Can you check if the order of declaration of student and staff abilities? I remember that there is a correct 'order' that we should follow. 

https://github.com/CanCanCommunity/cancancan/wiki/Ability-Precedence

**Comment by kxmbrian** [comment]
Correctness-wise, we only have to worry about the order if we use `cannot`s. `can`s are logically `OR`ed and `OR`s are commutative.

File: app/models/components/course/surveys_ability_component.rb [hunk 1]
```diff
@@ -13,59 +13,127 @@ def define_permissions
...
+      can :submit, Course::Survey::Response,
+          creator_id: user.id, submitted_at: nil, survey: ability_hash
+    end
+    can :submit, Course::Survey::Response, creator_id: user.id, submitted_at: nil,
+                                           survey: survey_expired_but_respondable
+  end
+
+  def allow_students_modify_own_response_to_active_survey
+    survey_active_all_course_users_hashes.each do |ability_hash|
+      can :modify, Course::Survey::Response,
```
Comment (allenwq) [inline app/models/components/course/surveys_ability_component.rb]: @kxmbrian Can put a comment why `modify` is used instead of `edit` ?

Comment (kxmbrian) [inline app/models/components/course/surveys_ability_component.rb]: Added above all the `modify` permissions

**Pull request merged**